devdoc/sm_requirements.md

```diff
@@ -0,0 +1,275 @@
+# State manager requirements
+================
+
+## Overview
+
+`State manager` (short:`sm`) is a module the manages the call state for the APIs of a module under the following sematics:
```

Comment:
>sm [](start = 24, length = 2)

One last attempt for state_mgr :-). If this fails I will just shut up about it :-)

Reply:
Cause it is poetry after all.

---
In reply to: [423408514](https://github.com/Azure/azure-c-util/pull/37#discussion_r423408514) [](ancestors = 423408514)

Reply:
I will create a separate task to rename it. Reason is I like is "sm" - it is very concise and it will live sooner or later in every single other module.

---
In reply to: [423408564](https://github.com/Azure/azure-c-util/pull/37#discussion_r423408564) [](ancestors = 423408564,423408514)

Reply:
Task 7203682: Rename "sm" to "state_manager" in sm.c in azure-c-util

---
In reply to: [423424447](https://github.com/Azure/azure-c-util/pull/37#discussion_r423424447) [](ancestors = 423424447,423408564,423408514)